core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -188,17 +189,20 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
       return lengths;
     }
 
-    final FileOutputStream out = new FileOutputStream(outputFile, true);
+    final FileChannel out = FileChannel.open(outputFile.toPath(),
+    ImmutableSet.of(WRITE, APPEND, CREATE));
     final long writeStartTime = System.nanoTime();
     boolean threwException = true;
     try {
       for (int i = 0; i < numPartitions; i++) {
         final File file = partitionWriterSegments[i].file();
         if (file.exists()) {
-          final FileInputStream in = new FileInputStream(file);
+          final FileChannel in = FileChannel.open(file.toPath(), ImmutableSet.of(READ));
           boolean copyThrewException = true;
           try {
-            lengths[i] = Utils.copyStream(in, out, false, transferToEnabled);
+            final long size = in.size();
```

Comment:
Why final? we generally don't write that unless it's required. I know the surrounding code does though anyway